spec/models/user_spec.rb

```diff
@@ -9,9 +9,28 @@
   end
 
   describe "validations" do
-    it { is_expected.to validate_presence_of(:first_name) }
-    it { is_expected.to validate_presence_of(:last_name) }
-    it { is_expected.to validate_presence_of(:email) }
+    subject { FactoryBot.create(:user) }
+    it { is_expected.to validate_presence_of(:first_name).with_message("Enter a first name") }
+    it { is_expected.to validate_presence_of(:last_name).with_message("Enter a last name") }
+    it { is_expected.to validate_presence_of(:email).with_message("Enter an email") }
+    it {
+      is_expected.to validate_uniqueness_of(:email)
+                       .case_insensitive
+                       .with_message("This email address is already in use")
+    }
+
+    it "rejects an invalid email" do
+      user = FactoryBot.create(:user)
```

Comment:
You could just do user = FactoryBot.build(:user, email: 'invalid'). Saves you having to do user.email = "invalid"

Reply:
That would be expecting the FactoryBot invocation to raise a RecordInvalid, which just seems wrong somehow

Reply:
FactoryBot.build(:user, email: 'invalid') won't cause an error. It's exactly what you need here as it will build the record but not automatically save it.

Reply:
Sorry, was using create 🙄